### ISSUE TITLE
xcodebuild hangs if run from command line not by flutter_tool

### ISSUE THREAD

**Issue by pulyaevskiy** [flutter/flutter#15642]
## Steps to Reproduce

I'm trying to integrate with fastlane but got blocked by a build command hanging forever. I based my `Fastlane` file on the [Gallery app](https://github.com/flutter/flutter/blob/master/examples/flutter_gallery/ios/fastlane/Fastfile) version.

The process stops on:

```ruby
    # Build and archive the app again.
    build_ios_app(
      workspace: 'Runner.xcworkspace',
      scheme: 'Runner',
      export_method: 'app-store',
    )
```

After trying to debug this I ended up trying to run following `xcodebuild` command:

```shell
# in ios subfolder
xcodebuild -workspace Runner.xcworkspace -scheme Runner \
  -destination 'generic/platform=iOS' \
  -archivePath /Users/username/Library/Developer/Xcode/Archives/2018-03-16/Runner\ 2018-03-16\ 20.41.22.xcarchive \
  archive \
  CODE_SIGN_IDENTITY=iPhone\ Distribution:\ My\ Name\ \(TEAM_ID\)
```

This command hangs on "Run script" step which executes (according to project config):

```shell
/bin/sh "$FLUTTER_ROOT/packages/flutter_tools/bin/xcode_backend.sh" build
```

> Note: if I try to execute this script directly it works. Only when it's run by xcodebuild it hangs.

With `export VERBOSE_SCRIPT_LOGGING=1` these are the last log entries I can see before everything stops:

```shell
♦ rm -rf -- /Users/anatoly/Projects/memspace/ios/Flutter/App.framework
♦ rm -f -- /Users/anatoly/Projects/memspace/ios/Flutter/app.flx
♦ cp -r -- /Users/anatoly/flutter/bin/cache/artifacts/engine/ios-release/Flutter.framework /Users/anatoly/Projects/memspace/ios/Flutter
♦ find /Users/anatoly/Projects/memspace/ios/Flutter/Flutter.framework -type f -exec chmod a-w {} ;
♦ /Users/anatoly/flutter/bin/flutter --suppress-analytics build aot --output-dir=build/aot --target-platform=ios --target=lib/main.dart --release --preview-dart-2
Building AOT snapshot in release mode (ios-release)...
Skipping AOT snapshot build. Fingerprint match.
Built to build/aot/.
^C** BUILD INTERRUPTED **
```

When this happens I can see a `dart` process running in the Activity Monitor, which seems to be idle.

I'm still not sure what can cause this. I was thinking it could be something wrong with environment on my machine (I suspect Gallery app builds are working fine?). But this hanging `dart` process looks very strange.

## Flutter Doctor

```shell
[✓] Flutter (Channel master, v0.2.3-pre.33, on Mac OS X 10.13.3 17D102, locale en-BY)
    • Flutter version 0.2.3-pre.33 at /Users/anatoly/flutter
    • Framework revision cbbf4d4b1b (4 hours ago), 2018-03-16 17:44:47 -0700
    • Engine revision 9671f63459
    • Dart version 2.0.0-dev.37.0.flutter-7328726088

[✓] Android toolchain - develop for Android devices (Android SDK 26.0.3)
    • Android SDK at /Users/anatoly/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-26, build-tools 26.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 9.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.2, Build version 9C40b
    • ios-deploy 1.9.2
    • CocoaPods version 1.3.1

[✓] Android Studio (version 3.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)

[✓] IntelliJ IDEA Community Edition (version 2017.3.4)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 22.2.2
    • Dart plugin version 173.4548.30

[✓] VS Code (version 1.21.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Dart Code extension version 2.10.0

[!] Connected devices
    ! No devices available

! Doctor found issues in 1 category.
```

**Comment by lukaspili** [comment]
I see the same issue with my fastlane script.
It was working fine last week, but not today anymore (upgrading to latest master several times per week).

**Comment by xster** [comment]
Can you guys help us debug this by opening `xcode_backend.sh` in the flutter repo and adding `set -e` at the beginning of `BuildApp()` function? 

It would help us figure out where it's stuck at.

**Comment by pulyaevskiy** [comment]
Thanks! I should have some time to try that over the weekend. Will update when I have something.

**Comment by lukaspili** [comment]
Did some quick checking and `set -e` does not provide more logs than what @pulyaevskiy posted.
BuildApp() function hangs on:

```
    RunCommand "${FLUTTER_ROOT}/bin/flutter" --suppress-analytics build aot \
      --output-dir="${build_dir}/aot"                                       \
      --target-platform=ios                                                 \
      --target="${target_path}"                                             \
      ${aot_flags}                                                          \
      ${local_engine_flag}                                                  \
      ${preview_dart_2_flag}
```

There is no errors, and I can also see a sleeping dart process until I kill the script manually.

**Comment by pulyaevskiy** [comment]
I can confirm what @lukaspili said. No difference in logs after adding `set -e`:

```shell
♦ /Users/anatoly/flutter/bin/flutter --suppress-analytics build aot --output-dir=build/aot --target-platform=ios --target=lib/main.dart --release --preview-dart-2
Building AOT snapshot in release mode (ios-release)...
Skipping AOT snapshot build. Fingerprint match.
Built to build/aot/.
```

Tried upgrading to latest master, still the same.

**Comment by bkonyi** [comment]
This should be fixed once #16116 lands. Already landed in the Dart SDK [here](https://github.com/dart-lang/sdk/commit/52afcba357ad398e8c24f3e3363ac6ff5293df63).
